packages/compiler/core/types.ts

```diff
@@ -1276,6 +1289,46 @@ export enum IdentifierKind {
   Other,
 }
 
+// Doc-comment related syntax
+
+export interface DocNode extends BaseNode {
+  readonly kind: SyntaxKind.Doc;
+  readonly content: readonly DocContent[];
+  readonly tags: readonly DocTag[];
+}
+
+export interface DocTagBaseNode extends BaseNode {
+  readonly tagName: IdentifierNode;
+  readonly content: readonly DocContent[];
+}
+
+export type DocTag = DocReturnsTagNode | DocParamTagNode | DocTemplateTagNode | DocUnknownTagNode;
+export type DocContent = DocTextNode;
+
+export interface DocTextNode extends BaseNode {
+  readonly kind: SyntaxKind.DocText;
+  readonly text: string;
+}
+
+export interface DocReturnsTagNode extends DocTagBaseNode {
+  readonly kind: SyntaxKind.DocReturnsTag;
+}
+
+export interface DocParamTagNode extends DocTagBaseNode {
+  readonly kind: SyntaxKind.DocParamTag;
+  readonly name: IdentifierNode;
```

Comment:
```suggestion
  readonly paramName: IdentifierNode;
```

would this be better to make it clearer what `tagName` vs `name` is?

same below I guess

Reply:
Changed.